src/ProductView/TabBar/QueryInterface/QueryInterfaceAttributeList/QueryInterfaceAttributeList.js

```diff
@@ -5,20 +5,34 @@ class QueryInterfaceAttributeList extends Component {
     constructor(props) {
         super(props);
         this.onChangeEventTypeName = this.onChangeEventTypeName.bind(this);
+        this.state = {
+            validationClasses: 'has-danger'
+        };
     }
 
     onChangeEventTypeName (event) {
         this.props.onChangeEventTypeName(event.target.value);
+        this.queryNameFormValidation(event.target.value);
+    }
+
+    queryNameFormValidation (queryName) {
+        if (!queryName) {
+            this.setState({ validationClasses: 'has-danger' });
+        }
+        else {
+            this.setState({ validationClasses: '' });
+        }
     }
 
     render() {
         return (
             <div className="query-interface-header">
-                <div className="form-group">
+                <div className={`form-group ` + this.state.validationClasses}>
                     <input className="form-control" type="text"
                            placeholder="Name" id="query-name"
                            value={this.props.eventTypeName}
                            onChange={this.onChangeEventTypeName}/>
+                    {this.state.validationClasses && <div className="form-control-feedback">This field can't be empty.</div>}
```

Comment:
This validation stuff occures the second time here. Please think about outsourcing it to a new component.